lib/index.es6

```diff
@@ -0,0 +1,97 @@
+import * as path from "path";
+import * as fs from "fs";
+
+import { watch } from "chokidar";
+import {sync as mkdirp} from "mkdirp";
+import most from "most";
+
+import compile from "./compile";
+import { entries } from "./util";
+import loadAst from "./compile/modules/load-ast";
+import compileModules from "./compile/modules/compile";
+
+export default function Interlock (options) {
+  const cwd = process.cwd();
+  options = options || {};
+
+  if (!options.emit || !options.emit.length) {
+    throw new Error("Must define at least one bundle.");
+  }
+
+  // TODO: validate options, or only persist specific values
+  options.context = options.context || cwd;
+  options.outputPath = options.outputPath || path.join(cwd, "dist");
+  options.extensions = options.extensions || [".js", ".jsx", ".es6"];
+  options.ns = options.ns || require(path.join(options.root, "./package.json")).name;
+  options.context = options.context || cwd;
+  this.options = options;
+}
+
+Interlock.prototype.build = function () {
+  compile(this.options).then(this._saveBundles);
+};
+
+Interlock.prototype._saveBundles = function (compilation) {
+  for (let [bundleDest, bundle] of entries(compilation.bundles)) {
+    const bundleOutput = bundle.raw;
+    const outputPath = path.join(compilation.opts.outputPath, bundleDest);
+    mkdirp(path.dirname(outputPath));
+    fs.writeFileSync(outputPath, bundleOutput);
```

Comment:
Nope.  There are some other areas that run synchronously, and those will receive a refactor.  Fortunately, it is easier now to see which pieces are synchronous and which are async, with `Pluggable.stream` `Pluggable.promise` and `Pluggable.sync` everywhere :)